src/cmdlib.sh

```diff
@@ -55,6 +55,10 @@ else
         *)         fatal "Architecture $(arch) not supported"
     esac
 fi
+
+# only use 1 core on kubernetes since we can't determine how much we can actually use
+grep -q 'kubepods' /proc/1/cgroup && procs=1 || procs="$(nproc)"
+QEMU_KVM+=" -smp $procs"
```

Comment:
I think we want to be explicit here as per [this thread](https://www.redhat.com/archives/libvirt-users/2017-June/msg00025.html)? So e.g. `-smp $procs,sockets=$procs,cores=1,threads=1` ? `qemu-kvm(1)` just says `Missing values will be computed.`, but doesn't really detail how they get computed.

Reply:
I figured that qemu would do the sensible thing. `qemu-kvm(1)` also says `For the PC target, the number of cores per socket, the number of threads per cores and the total number of sockets can be specified.` Which implies they _can't_ on other targets.

Empirically it looks like `-smp N` does in fact set the number of sockets on x86_64.

When running with `-smp 4`:
```
$ lscpu
...
Thread(s) per core:  1
Core(s) per socket:  1
Socket(s):           4
...
```

Reply:
Cool, thanks for checking!